src/pybreaker.py

```diff
@@ -106,21 +106,32 @@ def reset_timeout(self, timeout):
         """
         self._reset_timeout = timeout
 
+    def _create_new_state(self, state, notify=True):
+        """
+        Return state object from state string, i.e.,
+        'closed' -> <CircuitClosedState>
+        """
+        state_map = {
+            STATE_CLOSED: CircuitClosedState,
+            STATE_OPEN: CircuitOpenState,
+            STATE_HALF_OPEN: CircuitHalfOpenState,
+        }
+        try:
+            return state_map[state](self, state, notify=notify)
+        except KeyError:
+            raise ValueError(
+                "Unknown state (%r), valid states: %s",
+                state, ', '.join(state_map.keys()))
```

Comment:
I think you may need to go ahead and perform your string `format()` for the message argument. If this were a logger call, it would be a-ok.

Also, you can get away with `', '.join(state_map)`, as it iterates over the keys by default

Reply:
You're completely right on both parts.